plugins/default/lib/disk_usage.py

```diff
@@ -1,35 +1,97 @@
+check_disk_output_pattern = re.compile(
+    r"([A-Za-z0-9\./\-_]+)\s+(\d+)%\s+(\d+)%")
 
 
-def parse_df(line):
+def generate_parser():
+    parser = argparse.ArgumentParser()
+    parser.add_argument("-w", "--warn", type=int, action="store",
+                        required=True, help="Warning Threshold")
+    parser.add_argument('-c', '--crit', type=int, action="store",
+                        required=True, help="Critical Threshold")
+    return parser
+
+
+def parse_df_line(line):
     mo = check_disk_output_pattern.match(line)
     if mo is None:
         return ()
     return (mo.group(1), int(mo.group(2)), int(mo.group(3)))
 
-def analize(pcent, ipcent, warning_threshold, critical_threshold):
-    max_pcent = max(pcent, ipcent)
-    return max_pcent >= warning_threshold, max_pcent >= critical_threshold
 
-def report():
-    raise NotImplementedError("report is not implemented yet")
+def parse_df_lines(lines):
+    return (map(parse_df_line, lines.splitlines()[1:]))
+
+
+def analize(pod, disks, warning_threshold, critical_threshold):
+    results = []
+    for disk in disks:
+        max_pcent = max(disk[1], disk[2])
+
+        disk_status = nagios.UNKNOWN
+        if max_pcent >= critical_threshold:
+            disk_status = nagios.CRIT
+        elif max_pcent >= warning_threshold:
+            disk_status = nagios.WARN
+        elif max_pcent < warning_threshold:
+            disk_status = nagios.OK
+
+        results.append([pod, disk[0], disk[1], disk[2], disk_status])
+    return results
+
+
+def report(results):
+    counts = Counter(d[4] for d in results)
+
+    ret = max(counts)
+
+    if ret == nagios.OK:
+        print "%s: All %s volumes are under the warning threshold" % (nagios.status_code_to_label(ret), len(results))
+    elif ret == nagios.UNKNOWN:
+        print "%s: Unable to determine usage on %s volumes" % (nagios.status_code_to_label(ret), counts[nagios.UNKNOWN])
+    elif ret == nagios.WARN:
+        print "%s: There are %s volumes over the warning threshold" % (nagios.status_code_to_label(ret), counts[nagios.WARN])
+    else:
+        print "%s: There are %s volumes over the critical threshold and %s volumes over the warning threshold" % (nagios.status_code_to_label(ret), counts[nagios.CRIT], counts[nagios.WARN])
+
+    for r in results:
+        print "%s: %s:%s - bytes used: %s%%, inodes used: %s%%" % (nagios.status_code_to_label(r[4]), r[0], r[1], r[2], r[3])
+
+    return ret
+
 
 def main():
-    project = oc("project", "-q")
-    print analize(map(parse_df, exec_in_pods(project, get_running_pod_names(), check_disk_cmd)))
+    parser = generate_parser().parse_args()
+    warn = parser.warn
+    crit = parser.crit
+
+    if crit < warn:
+        raise ValueError("Critical threshold lower than Warning threshold")
+
+    project = openshift.get_project()
+
+    results = []
+
+    for p in openshift.exec_in_pods(project, openshift._get_running_pods(
```

Comment:
Sorry, that was from when I was debugging why the namespace filter wasn't working which I ended up finding was due to the new line at the end.